### ISSUE TITLE
Conversion of olca_type to json's @type not reliable

### ISSUE THREAD

**Issue by SebaJeku** [GreenDelta/olca-ipc.py#11]
With the schema.py commit 144f5c9 (Aug 14,2020), the initialization of an Entity's olca_type has been changed from None to ''.
So, without any further setting of the olca_type, a call to_json (e.g. via Client().insert()) results in @type left '' which -- at least in my environment with OpenLCA 1.10.3 as a server and Python 3.8 as an client -- results in JSON RPC error
   `{'code': -32602, 'message': 'Invalid params: params must be an object with valid @id and @type'}`
when running the "steel" flow example from https://pypi.org/project/olca-ipc/
In fact, after setting
  ` steel.olca_type = olca.schema.Flow.__name__`
it runs flawlessly.

**Comment by msrocka** [comment]
Could you check if this is still an issue when you install the version from the current master branch (e.g. via `pip install -U git+https://github.com/GreenDelta/olca-ipc.py.git/@master`)?

The `olca_type` field should be set in the constructor now. See also this test: https://github.com/GreenDelta/olca-ipc.py/blob/master/tests/test_model.py#L24

**Comment by gthoma** [comment]
Hi Michael,
I updated to the current master as described below.
I created a product system in olca and ran it with both 'as defined' and 'none' for the allocation type. I get different values for the impacts.
I then ran the following via IPC and the result returned was the same as the manual run with 'none' - despite using the olca.AllocationType.USE_DEFAULT_ALLOCATION
so the issue is not resolved.
Thanks
Greg

setup = olca.CalculationSetup()
setup.calculation_type = olca.CalculationType.SIMPLE_CALCULATION
setup.allocation_method = olca.AllocationType.USE_DEFAULT_ALLOCATION
# select the product system and LCIA method
im = client.find(olca.ImpactMethod, 'ReCiPe 2016 Midpoint (H)') #find can take string arguments
jl = client.get(olca.NwSet,'28298281-e344-33cf-818a-4b47371c47a3') #ReCiPe 2016 H, World (2010) H normalization.
setup.impact_method = olca.ref( olca.ProductSystem, im.id ) #ref returns a pointer; takes UUID argument
setup.product_system = client.get(olca.ProductSystem, prodSys.id) #get returns the object; takes UUID argument
setup.nw_set = jl

setup.amount = 1000.0
# calculate the result and export it to an Excel file
result1 = client.calculate(setup)
client.excel_export(result1, farm.Name + 'tst3.xlsx')

**Comment by msrocka** [comment]
Hi Greg,

(this was another issue: #11 instead of #7)

I created this test to check this: https://github.com/GreenDelta/olca-ipc.py/blob/master/tests/test_allocation.py#L59

With the current test version of openLCA 2.0 from https://share.greendelta.com/index.php/s/2AJJMhfB1WSIFDO this works for me and a I get a different result for each allocation method. Does this test work for you?

Thanks,
Michael

@gthoma @ju1ianr

**Comment by ezrakahn** [comment]
I am finding a similar issue regarding model_type.  the attribute is coming in empty, '', both when I create a new model such as 

new = olca.Flow()

or when I query one with ipc for example:

flow = client.get(olca.Flow, refid)

The returned olca.schema.flow object is missing an object type. I tried this with processes and categories too, with same behavior.  I have the latest 0.0.10 package.


Sorry, I imported latest master commit and this problem is fixed.
